src/sqlfluff/dialects/dialect_postgres.py

```diff
@@ -2322,6 +2324,7 @@ class TableConstraintUsingIndexSegment(BaseSegment):
     As specified in: https://www.postgresql.org/docs/13/sql-altertable.html.
     """
 
+    type = "table_constraint_segment"
```

Comment:
```suggestion
    type = "table_constraint"
```

Reply:
This is a bit bigger as already set to `table_constraint_segment` in other dialects. Agree we should change it though. But don't be surprised when even more YAMLs change now!

Reply:
Done.